From c4e96a305fbca4f83ae9f3a81482481524380905 Mon Sep 17 00:00:00 2001
From: NeilBrown <neilb@suse.de>
Date: Mon, 20 Aug 2012 10:37:21 +1000
Subject: mdmon: allow --takeover when original was started with --offroot
Bug-Debian: http://bugs.debian.org/691671
Comment: from stable/bugfix upstream 3.2.6 version
Comment: one of the 3 patches fixing mentioned issues

As --offroot causes ARGV[0] to be changed, we need to be more
lenient when checking that the mdmon we are about to kill really
is mdmon.  i.e. allow name to be "@dmon" instead.

Signed-off-by: NeilBrown <neilb@suse.de>
---
 mdmon.c |    6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/mdmon.c b/mdmon.c
index 2d0503d..54e1238 100644
--- a/mdmon.c
+++ b/mdmon.c
@@ -184,7 +184,11 @@ static void try_kill_monitor(pid_t pid, char *devname, int sock)
 	buf[sizeof(buf)-1] = 0;
 	close(fd);
 
-	if (n < 0 || !strstr(buf, "mdmon"))
+	/* Note that if started with --offroot, the name
+	 * might be "@dmon"
+	 */
+	if (n < 0 || !(strstr(buf, "mdmon") ||
+		       strstr(buf, "@dmon")))
 		return;
 
 	kill(pid, SIGTERM);
-- 
1.7.10.4

